--- java-gcj-compat-1.0.76/rebuild-gcj-db.in	2007-03-26 18:13:29.000000000 +0300
+++ java-gcj-compat-1.0.76/rebuild-gcj-db.in.n	2007-06-23 00:29:55.000000000 +0300
@@ -11,6 +11,13 @@
    fi
    dirname $dbLocation | xargs mkdir -p
    @GCJ_BIN_DIR@/gcj-dbtool@gcc_suffix@ -n $dbLocation 64
-   find $libdir -follow -name '*.db' -print0 | \
+   find $libdir -follow -name '*.db' | {
+      while read dbfile; do
+         sofile="${dbfile%.db}.so"
+         if [ -e "$sofile" ] && [ 0@GCJ_BC_MAJOR@ -ne 0$(ldd "$sofile" | sed -ne 's,^.*libgcj_bc.so.\([^ ]\).*$,\1,p') ]; then
+             continue
+         fi
+         echo -ne "${dbfile}\x00"
+      done; } |
       xargs -0 @GCJ_BIN_DIR@/gcj-dbtool@gcc_suffix@ -m $dbLocation $dbLocation
 done
