--- java-gcj-compat-1.0.68/aotcompile.py.in	2006-09-08 00:59:16.000000000 +0300
+++ java-gcj-compat-1.0.68-f/aotcompile.py.in	2007-06-24 16:26:36.000000000 +0300
@@ -154,8 +154,8 @@
     # option is temporary in aot-compile-rpm and dead options left in
     # specfiles will hinder its removal.
     for path in exclusions:
-        if path in paths:
-            paths.remove(path)
+        if os.path.normpath(path) in paths:
+            paths.remove(os.path.normpath(path))
         else:
             raise Error, "%s: path does not exist or is not a job" % path
     # Build the list of jobs
